sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -2321,18 +2410,23 @@ def __getitem__(self, name):
         self._grouping_indexes,
         projection=name)
 
-  def agg(self, fn):
-    if not callable(fn):
-      # TODO: Add support for strings in (UN)LIFTABLE_AGGREGATIONS. Test by
-      # running doctests for pandas.core.groupby.generic
-      raise NotImplementedError('GroupBy.agg currently only supports callable '
-                                'arguments')
-    return DeferredDataFrame(
-        expressions.ComputedExpression(
-            'agg',
-            lambda gb: gb.agg(fn), [self._expr],
-            requires_partition_by=partitionings.Index(),
-            preserves_partition_by=partitionings.Singleton()))
+  def agg(self, fn, *args, **kwargs):
+    if callable(fn):
```

Comment:
Do we want things like np.sum to take this path, or should we put this case last?

Reply:
Good catch, I moved this to the end. It also exposed an edge case in preagg proxy generation. I pushed ddd0885 to address this./